ports/esp32/machine_pwm.c

```diff
@@ -196,87 +204,72 @@ void machine_pwm_deinit_all(void) {
 STATIC void configure_channel(machine_pwm_obj_t *self) {
     ledc_channel_config_t cfg = {
         .channel = self->channel,
-        .duty = (1 << (timers[TIMER_IDX(self->mode, self->timer)].duty_resolution)) / 2,
+        .duty = self->channel_duty,
```

Comment:
This is the most important change in this PR, and should probably be it's own standalone fix?  (possibly along with whatever set channel_duty up properly?  This what prevents the pwm from glitching at setup, before the user specified duty is finally applied

Reply:
I implemented something similar, just this initial glitch fix in https://github.com/micropython/micropython/pull/12650